module/VuFindSearch/src/VuFindSearch/Backend/EDS/SearchRequestModel.php

```diff
@@ -211,6 +211,27 @@ public function setParameters($parameters = [])
         }
     }
 
+    /**
+     * Checks whether the search model is valid for the EDS API.
+     *
+     * @return bool Returns false if any model parameters are invalid.
+     */
+    public function isValid()
+    {
+        $contentProviderValues = $this->facetFilters['ContentProvider'] ?? [];
+        foreach ($contentProviderValues as $value) {
+            // Content providers should be only alphanumeric+space strings.
+            // This explode + ctype logic is about 8x faster than using a regex.
```

Comment:
Specifically in testing on a VM on my laptop, I was generally seeing 5ms speeds with ctype, vs around 40ms with `preg_match('/[a-zA-Z0-9\s]+/', $value);`

Reply:
Moot, removed.  But the new code is also ~5ms.